osquery/events/linux/bpf/bpfeventpublisher.cpp

```diff
@@ -102,6 +106,16 @@ Status BPFEventPublisher::setUp() {
     return Status::failure("Publisher disabled via configuration");
   }
 
+  // By default, VMware Fusion will report 128 possible processors. This
+  // causes a huge increase in memory usage when using per-cpu maps. Use
+  // low settings as default to be safe, but inform the user about this.
+  if (FLAGS_bpf_buffer_storage_size == kDefaultBufferStorageSize ||
+      FLAGS_bpf_perf_event_array_exp == kDefaultPerfEventArrayExp) {
+    LOG(INFO)
```

Comment:
I do not think we should have this `INFO` message on by default to warn folks of this edge case. I recommend limiting this message by first detecting the values of the sysfs entries.

Reply:
Also heads up you should be able to use the `osquery/core/flags.h` APIs to check if flag values are set to their default, meaning the user or config did not try to overwrite them.

Reply:
I think I am starting to change my mind on this; maybe it's better to always use low values and update the docs to suggest the user to tweak the values when deploying to production machines? The value set here depends too much on how many cores are present on the machine, and they don't always scale in a predictable way vs the available memory